lib/rihanna/job.ex

```diff
@@ -177,6 +177,77 @@ defmodule Rihanna.Job do
   end
 
   @doc false
+  def delete_by(opts) do
+    ids_to_delete =
+      opts
+      |> filter_term_list()
+      |> Enum.join(",")
+
+    if ids_to_delete != "" do
+      case producer_query(
+             """
+                DELETE FROM "#{table()}"
+                WHERE id IN (#{ids_to_delete})
+             """,
+             []
+           ) do
+        {:ok, %Postgrex.Result{num_rows: 0}} ->
+          {:error, :job_not_found}
+
+        {:ok, _} ->
+          {:ok, :deleted}
+
+        error ->
+          error
+      end
+    else
+      {:error, :job_not_found}
+    end
+  end
+
+  defp filter_term_list(mod: mod, fun: fun) when not is_nil(mod) and not is_nil(fun) do
+    Enum.flat_map(retrieve_all_jobs(), fn [id, term] ->
+      if match?({^mod, ^fun}, :erlang.binary_to_term(term)) do
+        [id]
+      else
+        []
+      end
+    end)
+  end
+
+  defp filter_term_list(mod: mod) when not is_nil(mod) do
+    Enum.flat_map(retrieve_all_jobs(), fn [id, term] ->
+      if match?({^mod, _}, :erlang.binary_to_term(term)) do
+        [id]
+      else
+        []
+      end
+    end)
+  end
+
+  defp filter_term_list(fun: fun) when not is_nil(fun) do
+    Enum.flat_map(retrieve_all_jobs(), fn [id, term] ->
+      if match?({_, ^fun}, :erlang.binary_to_term(term)) do
```

Comment:
```suggestion
      if match?({_, ^fun}, :erlang.binary_to_term(binary)) do
```

If converts binary to atom the param should be a binary.

https://hexdocs.pm/elixir/String.html#to_atom/1 it works in this case?

Reply:
It is binary, it is stored using `erlang.term_to_binary/1`, it's just a misnaming.